governance/xc_admin/packages/xc_admin_frontend/components/tabs/UpdateProductMetadata.tsx

```diff
@@ -0,0 +1,346 @@
+import toast from 'react-hot-toast'
+import { getMultisigCluster, proposeInstructions } from 'xc_admin_common'
+import { ClusterContext } from '../../contexts/ClusterContext'
+import { usePythContext } from '../../contexts/PythContext'
+import { SECURITY_MULTISIG, useMultisig } from '../../hooks/useMultisig'
+import { capitalizeFirstLetter } from '../../utils/capitalizeFirstLetter'
+import ClusterSwitch from '../ClusterSwitch'
+import Modal from '../common/Modal'
+import Spinner from '../common/Spinner'
+import Loadbar from '../loaders/Loadbar'
+
+interface SymbolToProductMetadata {
+  [key: string]: Product
+}
+
+interface ProductMetadataInfo {
+  prev: Product
+  new: Product
+}
+
+const symbolToProductAccountKeyMapping: Record<string, PublicKey> = {}
+
+const UpdateProductMetadata = () => {
+  const [data, setData] = useState<SymbolToProductMetadata>({})
+  const [productMetadataChanges, setProductMetadataChanges] =
+    useState<Record<string, ProductMetadataInfo>>()
+  const [isModalOpen, setIsModalOpen] = useState(false)
+  const [isSendProposalButtonLoading, setIsSendProposalButtonLoading] =
+    useState(false)
+  const { cluster } = useContext(ClusterContext)
+  const anchorWallet = useAnchorWallet()
+  const { isLoading: isMultisigLoading, squads } = useMultisig(
+    anchorWallet as Wallet
+  )
+  const { rawConfig, dataIsLoading, connection } = usePythContext()
+  const { connected } = useWallet()
+  const [pythProgramClient, setPythProgramClient] =
+    useState<Program<PythOracle>>()
+
+  const openModal = () => {
+    setIsModalOpen(true)
+  }
+
+  const closeModal = () => {
+    setIsModalOpen(false)
+  }
+
+  useEffect(() => {
+    if (!dataIsLoading && rawConfig && rawConfig.mappingAccounts.length > 0) {
+      const symbolToProductMetadataMapping: SymbolToProductMetadata = {}
+      rawConfig.mappingAccounts
+        .sort(
+          (mapping1, mapping2) =>
+            mapping2.products.length - mapping1.products.length
+        )[0]
+        .products.map((product) => {
+          symbolToProductAccountKeyMapping[product.metadata.symbol] =
+            product.address
+          symbolToProductMetadataMapping[product.metadata.symbol] =
+            product.metadata
+        })
+      setData(sortData(symbolToProductMetadataMapping))
+    }
+  }, [rawConfig, dataIsLoading])
+
+  const sortData = (data: SymbolToProductMetadata) => {
+    const sortedSymbolToProductMetadataMapping: SymbolToProductMetadata = {}
+    Object.keys(data)
+      .sort()
+      .forEach((key) => {
+        const sortedInnerData: any = {}
+        Object.keys(data[key])
+          .sort()
+          .forEach((innerKey) => {
+            sortedInnerData[innerKey] = data[key][innerKey]
+          })
+        sortedSymbolToProductMetadataMapping[key] = sortedInnerData
+      })
+
+    return sortedSymbolToProductMetadataMapping
+  }
+
+  // function to download json file
+  const handleDownloadJsonButtonClick = () => {
```

Comment:
Two fields in the downloaded json are not really editable price_account and symbol.
I suggest you don't show price_account (low-level info).
Symbol is already there as it is the key, so you can get away with not showing.

If someone tries to edit/add these fields, raise a toast error.